proposals/Integrate with external device inventory.md

```diff
@@ -0,0 +1,44 @@
+
+The above issues can be resolved by allowing Akri getting additional device infomration from an external optional device inventory system. This proposal will introduce a few additional steps in Akri's workflow to allow Akri extracting addtioanl device information from external source. 
+
+<br/><br/>
+
+## Proposal
+
+The below workflow presents a new flow from discovering device till creating broker POD. Additional steps are highlighted as yellow color.
+<img src="../media/integrate-extenal-device-inventory-diagram.svg" style="padding-bottom: 10px padding-top: 20px; margin-right: auto; display: block; margin-left: auto;background-color:white"/>
+
+- Introduce a new setting that user can specify the name of a customized external device inventory POD or service, so Akri agent can request for additional device information. The proposed external device inventory POD/service must expose a REST POST API /queryDevice for Akri Agent to raise request.
+- Instead of directly creating Akri instance after Akri agent discovers new devices, Akri agent will query the external device inventory POD/service to get more device related information, including credential. 
+- Please note for protected ONVIF device, Akri agent should use the ONVIF credential to finish a successful discovery work, such as querying stream URI.
+
+This proposed external device inventory POD/service name should be set within each Akri configuration, so it is possible that different discovery config has specified external device inventory system. Another flag is proposed to define if the related external device inventory sytem is also device provisioning database. If it is provisioning database, when Akri agent can not fetch any record of a discovered device, it wont create Akri instance for this device. There wont be broker POD created neither for that device. 
+
+The external device inventory POD is not deployed by Akri but solution developer will take care. One possible implementation of the device inventory POD can be a POD with a mounted Azure Key Vault drive using AKV ARC extenstion. It can fetch device credential and assist Akri agent. The benefit is that device credential can be centrally managed from cloud.   
+
+<br/><br/>
+
+## Additional Usage 1: what if a device is undiscoverable but can be added manually
+With this new proposed connection between Akri Agent and external device inventory system, Akri can achieve one more new feature. Akri can create broker POD for an undiscoverable device, such case like a camera does not support ONVIF discovery but we know its RTSP stream URI.
+To achieve this, Akri Agent may provide a HTTP POST REST API with URI /addDevice. External inventory system may trigger a manual device adding procedure with all necessary device infomation. After receiving "/addDevice" request, Akri agent can still create a Akrii and in the end a broker POD is created to process this manually added device.
```

Comment:
instead of Akri hosting the service, maybe it would be better to add another endpoint to the inventory service that should also be exposed to kubernetes. This matches akris approach to date of query for devices rather than being told about them.

Reply:
but if we want the `updateDevice` endpoint as well, this may make sense